app/lib/dags/experimental.json.jbuilder

```diff
@@ -0,0 +1,164 @@
+    end
+  end
+end
+
+json.steps do
+  steps = []
+
+  steps << {
+    in: ["taxid_fasta_in"],
+    out: "taxid_fasta_out",
+    class: "PipelineStepGenerateTaxidFasta",
+    module: "idseq_dag.steps.generate_taxid_fasta",
+    additional_files: {
+      lineage_db: attr[:lineage_db],
+    },
+    additional_attributes: {},
+  }
+
+  steps << {
+    in: ["taxid_fasta_out"],
+    out: "taxid_locator_out",
+    class: "PipelineStepGenerateTaxidLocator",
+    module: "idseq_dag.steps.generate_taxid_locator",
+    additional_files: {},
+    additional_attributes: {},
+  }
+
+  steps << {
+    in: [
+      "gsnap_m8",
+      "taxid_locator_out",
+    ],
+    out: "alignment_viz_out",
+    class: "PipelineStepGenerateAlignmentViz",
+    module: "idseq_dag.steps.generate_alignment_viz",
+    additional_files: {
+      nt_loc_db: attr[:nt_loc_db],
+    },
+    additional_attributes: {
+      nt_db: attr[:nt_db],
+    },
+  }
+
+  steps << {
+    in: ["fastqs"],
+    out: "srst2_out",
+    class: "PipelineStepRunSRST2",
+    module: "idseq_dag.steps.run_srst2",
+    additional_files: {
+      resist_gene_db: "s3://idseq-database/amr/ARGannot_r2.fasta",
+      resist_genome_bed: "s3://idseq-database/amr/argannot_genome.bed",
+    },
+    additional_attributes: {
+      min_cov: 0,
+      n_threads: 16,
+      file_ext: attr[:file_ext],
+    },
+  }
+
+  steps << {
+    in: ["refined_gsnap_in", "contig_in", "gsnap_m8"],
+    out: "coverage_viz_out",
+    class: "PipelineStepGenerateCoverageViz",
+    module: "idseq_dag.steps.generate_coverage_viz",
+    additional_files: {
+      info_db: "s3://idseq-database/alignment_data/2018-12-01/nt_info.sqlite3",
+    },
+    additional_attributes: {},
+  }
+
+  steps << {
+    in: ["fastqs", "nonhost_fasta"],
+    out: "nonhost_fastq_out",
+    class: "PipelineStepNonhostFastq",
+    module: "idseq_dag.steps.nonhost_fastq",
+    additional_files: {},
+    additional_attributes: {},
+  }
+
+  json.array! steps
+end
+
+json.given_targets do
+  json.taxid_fasta_in do
+    json.s3_dir "s3://#{attr[:bucket]}/samples/#{attr[:project_id]}/#{attr[:sample_id]}/results/#{attr[:pipeline_version]}"
+  end
+
+  json.gsnap_m8 do
+    json.s3_dir "s3://#{attr[:bucket]}/samples/#{attr[:project_id]}/#{attr[:sample_id]}/results/#{attr[:pipeline_version]}"
+  end
+
+  json.refined_gsnap_in do
+    json.s3_dir "s3://#{attr[:bucket]}/samples/#{attr[:project_id]}/#{attr[:sample_id]}/postprocess/#{attr[:pipeline_version]}/assembly"
+  end
+
+  json.contig_in do
+    json.s3_dir "s3://#{attr[:bucket]}/samples/#{attr[:project_id]}/#{attr[:sample_id]}/postprocess/#{attr[:pipeline_version]}/assembly"
+  end
+
+  json.fastqs do
+    json.s3_dir "s3://#{attr[:bucket]}/samples/#{attr[:project_id]}/#{attr[:sample_id]}/fastqs"
+  end
+
+  json.nonhost_fasta do
+    json.s3_dir "s3://#{attr[:bucket]}/samples/#{attr[:project_id]}/#{attr[:sample_id]}/postprocess/#{attr[:pipeline_version]}/assembly"
+  end
```

Comment:
I find this a little harder to read than
```
nonhost_fasta: {
  s3_dir: "s3://...",
}
```
The jbuilder files do feel a little harder to read than the erb ones. It's harder to see where the object keys are at a glance and it feels a little more verbose. Anything you can do to organize the file better would be appreciated, but it's not urgent.

Reply:
Unfortunately it won't let me change the nested hash to just:
```
json.nonhost_fasta { s3_dir: "s3://..." }
```
but I find this syntax about the same after getting used to it.

On balance I think the jbuilder construction is cleaner and avoids constructs like:
```
      "in": [
        <% @attribute_dict[:taxon_byteranges].keys.each_with_index do |pipeline_run_id, idx| %>
          "prepare_taxon_fasta_<%= pipeline_run_id %>_out"
          <% unless idx == @attribute_dict[:taxon_byteranges].size - 1 %>
            ,
          <% end %>
        <% end %>
      ],
```
```
    "host_filter_out": [
      "gsnap_filter_1.fa"
      <% if @attribute_dict[:input_file_count] > 1 %>
        , "gsnap_filter_2.fa"
        , "gsnap_filter_merged.fa"
      <% end %>
    ],
```
and inconsistent formatting in the generated files (as well as the main benefits of string handling). For overall verbosity I think jbuilder could be even more DRY in re-using variables, but I'm hearing that people want the "at a glance" template style as you do.